### PR TITLE
Event mails only to selected year students

### LINKED ISSUE #166: Send event mails only to users of the years the event is targeted to
While creating a new event, there should be an option to select which Academic Year students receive the mail. Currently, the mail is being sent to all the registered users.
Eg. 2023, 2024:  So both year students will receive the mail and not others.

### PR DESCRIPTION
**Pull request by RaviMauryaHootowl**
Fixes #166 

When adding a new event, the admin gets the option to select a range of Graduation Years. Initial Event mails are sent only to those students. 
Later, any student irrespective of their Grad Year can register and receive notification for that before the event starts.

(also, if we don't want to send mails to anyone, just put any bigger year like 2030, 2030 😂)

image

Auto Refresh Token is also implemented for Events(Create, Update, Delete) using the `useAuthenticatedAxios` Hook made in the last PR.

There are also slight schema changes for the database, graduationYear is replaced by graduationYearFrom and graduationYearTo in Events.
There doesn't seem any conflicts with previous data with the old schema as this graduationYearFrom and graduationYearTo fields are only used for sending emails initially.

If there are any changes, do let me know. Thanks!

### CODE REVIEW

File: new_client/src/components/events/AddEvent.jsx [hunk 1]
```diff
@@ -260,24 +264,32 @@ function AddEvent(props) {
                 </Grid>
               </Grid>
             </div>
-
+            Graduation Year Range: (For Event mails)
             <div className="form-group">
               <Grid container>
-                <Grid item xs={12}>
+                <Grid item xs={30}>
```
Comment (Saif807380) [inline new_client/src/components/events/AddEvent.jsx:270]: I think 12 is the maximum value

File: new_client/src/components/events/AddEvent.jsx [hunk 2]
```diff
@@ -260,24 +264,32 @@ function AddEvent(props) {
...
+                  <TextField
+                    type="text"
+                    placeholder="From Graduation year"
+                    name="graduationYearFrom"
+                    value={eventGraduationYearFrom}
+                    onChange={(e) => setEventGraduationYearFrom(e.target.value)}
+                    required
+                    label="From Graduation year"
```
Comment (Saif807380) [inline new_client/src/components/events/AddEvent.jsx:278]: g should be lowercase

File: new_client/src/components/events/AddEvent.jsx [hunk 3]
```diff
@@ -260,24 +264,32 @@ function AddEvent(props) {
...
+                  />
                   <TextField
```
Comment (Saif807380) [inline new_client/src/components/events/AddEvent.jsx:280]: Can we add a little spacing between the two textfields?

File: new_client/src/components/events/AddEvent.jsx (hunk 1 above)
Comment (Saif807380) [inline new_client/src/components/events/AddEvent.jsx:270]: I think 12 is the maximum value

File: new_client/src/components/events/AddEvent.jsx (hunk 2 above)
Comment (Saif807380) [inline new_client/src/components/events/AddEvent.jsx:278]: g should be lowercase

File: new_client/src/components/events/AddEvent.jsx (hunk 3 above)
Comment (Saif807380) [inline new_client/src/components/events/AddEvent.jsx:280]: Can we add a little spacing between the two textfields?

**Review by Saif807380 — CHANGES_REQUESTED** [review]

**Comment by RaviMauryaHootowl** [comment]
Made the above changes and also shifted verifyToken from AuthController to auth middleware.

**Review by Saif807380 — APPROVED** [review]

**Review by ShubhankarKG — APPROVED** [review]

**Pull request merged**